calico_containers/tests/unit/datastore_test.py

```diff
@@ -1211,6 +1212,136 @@ def test_get_default_node_as_no_config(self):
 
         assert_equal(self.datastore.get_default_node_as(), 64511)
 
+    def test_get_hostnames(self):
+        """
+        Test get_hostnames()
+        """
+        def mock_read(path):
```

Comment:
Oh lordy - so you mean that a single etcd read at the top level will return the full tree even if recursive is False?